### PR TITLE
dhruv/MT5Defaulting/password-modal-styling/TRAH-3963

### PR DESCRIPTION
**Pull request by Dhruv-deriv**
## Changes:

1. To style the new password modal in MT5-new Account creation 
2. To create the Mobile view for the new MT5 Account Creation Password model.
3. Created test cases for Mt5CreatePassword Modal

## Screenshot:

1. Desktop view:

Screenshot 2024-08-15 at 8 50 31 AM

2. Mobile view:
Screenshot 2024-08-23 at 2 10 30 PM

3. Test cases 
Screenshot 2024-08-23 at 2 41 02 PM

### CODE REVIEW

File: packages/cfd/src/sass/mt5-passwordmodal.scss [hunk 1]
```diff
@@ -0,0 +1,226 @@
+.dc-modal__container_cfd-password-modal,
```
Comment (amina-deriv) [inline packages/cfd/src/sass/mt5-passwordmodal.scss]: @Dhruv-deriv lets move this file inside mt5-password-modal folder

File: packages/cfd/src/sass/mt5-passwordmodal.scss [hunk 2]
```diff
@@ -0,0 +1,226 @@
...
+.dc-mobile-dialog__cfd-password-modal {
+    display: flex;
+    flex-direction: column;
+    justify-content: flex-start;
+    max-width: 424px;
```
Comment (amina-deriv) [inline packages/cfd/src/sass/mt5-passwordmodal.scss]: lets use rem

File: packages/cfd/src/sass/mt5-passwordmodal.scss [hunk 3]
```diff
@@ -0,0 +1,226 @@
...
+        align-items: flex-start;
+        justify-content: space-between;
+        padding: 0 2.4rem 1.2rem;
+
+        @include mobile-or-tablet-screen {
+            padding: 0;
+        }
+    }
+    &__description {
+        @include typeface(--small-center-normal-black);
```
Comment (amina-deriv) [inline packages/cfd/src/sass/mt5-passwordmodal.scss]: do we really need this?

File: packages/cfd/src/sass/mt5-passwordmodal.scss [hunk 4]
```diff
@@ -0,0 +1,226 @@
...
+        color: var(--text-prominent);
```
Comment (amina-deriv) [inline packages/cfd/src/sass/mt5-passwordmodal.scss]: we can pass this color as prop to Text component

File: packages/cfd/src/sass/mt5-passwordmodal.scss (hunk 1 above)
Comment (Dhruv-deriv) [reply packages/cfd/src/sass/mt5-passwordmodal.scss]: Done 👍

File: packages/cfd/src/sass/mt5-passwordmodal.scss (hunk 2 above)
Comment (Dhruv-deriv) [reply packages/cfd/src/sass/mt5-passwordmodal.scss]: Done 👍

File: packages/cfd/src/sass/mt5-passwordmodal.scss (hunk 3 above)
Comment (Dhruv-deriv) [reply packages/cfd/src/sass/mt5-passwordmodal.scss]: removed Done 👍

File: packages/cfd/src/sass/mt5-passwordmodal.scss [hunk 5]
```diff
@@ -0,0 +1,226 @@
...
+        &-title {
+            margin-top: 2.4rem;
+            margin-bottom: 0.8rem;
+        }
+        &-description {
+            max-width: 30rem;
+            margin-bottom: 1.6rem;
+        }
+    }
+    &__change-password-confirmation {
```
Comment (amina-deriv) [inline packages/cfd/src/sass/mt5-passwordmodal.scss]: remove this

Comment (Dhruv-deriv) [reply packages/cfd/src/sass/mt5-passwordmodal.scss]: done 👍

File: packages/cfd/src/sass/mt5-passwordmodal.scss (hunk 3 above)
Comment (Dhruv-deriv) [reply packages/cfd/src/sass/mt5-passwordmodal.scss]: doesnt break anything have removed it

File: packages/cfd/src/sass/mt5-passwordmodal.scss (hunk 2 above)
Comment (Dhruv-deriv) [reply packages/cfd/src/sass/mt5-passwordmodal.scss]: done 👍

File: packages/cfd/src/sass/mt5-passwordmodal.scss (hunk 4 above)
Comment (Dhruv-deriv) [reply packages/cfd/src/sass/mt5-passwordmodal.scss]: done 👍

File: packages/cfd/src/Containers/mt5-create-password/mt5-create-password.scss [hunk 6]
```diff
@@ -0,0 +1,200 @@
...
+        flex-grow: 1;
+        margin: auto;
+
+        @include mobile-or-tablet-screen {
+            width: calc(100vw - 4.8rem);
+            max-width: 30rem;
+        }
+    }
+    & .mt5-password-field {
+        margin-bottom: 1em;
```
Comment (amina-deriv) [inline packages/cfd/src/Containers/mt5-create-password/mt5-create-password.scss]: why not rem?

File: packages/cfd/src/Containers/mt5-create-password/mt5-create-password.scss [hunk 7]
```diff
@@ -0,0 +1,200 @@
...
+        }
+    }
+    @include tablet-up {
+        .cfd-password-modal__content {
+            display: flex;
+            flex-direction: column;
+            width: 100%;
+            margin-bottom: 2rem;
+
+            &--password-reset {
```
Comment (amina-deriv) [inline packages/cfd/src/Containers/mt5-create-password/mt5-create-password.scss]: why do we need this?

File: packages/cfd/src/Containers/mt5-create-password/mt5-create-password.scss [hunk 8]
```diff
@@ -0,0 +1,200 @@
...
+            padding: 0.8rem;
+            background-color: var(--status-warning);
+        }
+    }
+    &__message {
+        max-width: 32rem;
+        margin: auto;
+        line-height: 1.43;
+    }
+    &__radio {
```
Comment (amina-deriv) [inline packages/cfd/src/Containers/mt5-create-password/mt5-create-password.scss]: where is this used?

File: packages/components/src/components/modal/modal.scss [hunk 9]
```diff
@@ -1,5 +1,10 @@
 /** @define dc-modal; weak */
 .dc-modal {
+    @include mobile-screen {
+        width: 100%;
+        height: 100%;
+    }
```
Comment (amina-deriv) [inline packages/components/src/components/modal/modal.scss:6]: This will effect all the modals. Please revert this change

File: packages/components/src/components/modal/modal.tsx [hunk 10]
```diff
@@ -149,8 +149,8 @@ const ModalElement = ({
                 'dc-modal__container--is-confirmation-modal': is_confirmation_modal,
             })}
             style={{
-                height: height || 'auto',
-                width: width || 'auto',
+                height: height || '100%',
```
Comment (amina-deriv) [inline packages/components/src/components/modal/modal.tsx:152]: Please revert this changes as it will effect all components

File: packages/cfd/src/Containers/mt5-create-password/mt5-create-password.scss (hunk 6 above)
Comment (Dhruv-deriv) [reply packages/cfd/src/Containers/mt5-create-password/mt5-create-password.scss]: done 👍

File: packages/cfd/src/Containers/mt5-create-password/mt5-create-password.scss (hunk 7 above)
Comment (Dhruv-deriv) [reply packages/cfd/src/Containers/mt5-create-password/mt5-create-password.scss]: done 👍

File: packages/cfd/src/Containers/mt5-create-password/mt5-create-password.scss (hunk 8 above)
Comment (Dhruv-deriv) [reply packages/cfd/src/Containers/mt5-create-password/mt5-create-password.scss]: done 👍
